helix-rest/src/main/java/org/apache/helix/rest/metadatastore/ZkMetadataStoreDirectory.java

```diff
@@ -170,20 +170,21 @@ public void refreshRoutingData(String namespace) {
     // Check if namespace exists; otherwise, return as a NOP and log it
     if (!_routingZkAddressMap.containsKey(namespace)) {
       LOG.error("Failed to refresh internally-cached routing data! Namespace not found: " + namespace);
+      return;
     }
 
     try {
-      _realmToShardingKeysMap.put(namespace, _routingDataReaderMap.get(namespace).getRoutingData());
+      Map<String, List<String>> routingData = _routingDataReaderMap.get(namespace).getRoutingData();
+      _realmToShardingKeysMap.put(namespace, routingData);
```

Comment:
Null check  on _realmToShardingKeysMap since method is not necessarily threadsafe.

Reply:
As a matter of fact, let's do this check right in the beginning of this method.

If any of the maps (_routingZkAddressMap, _routingZkAddressMap, _routingDataMap) are null, just return.

Reply:
Question: if the method is not threadsafe, is a null check at beginning really enough? For example, by the time it gets to `_routingDataMap.put()`, `_routingDataMap` can still be null, same thing applies to `_realmToShardingKeysMap`. 
On the other hand, all map fields are final and they are initialized to be not null. In what situations will they actually become null?

Reply:
Interesting Java gotcha! Thank you for explaining offline @narendly !